### PR TITLE
SHARE template first cut - revised

### PR DESCRIPTION
**Pull request by waidyanatha**
A new PR after applying corrections requested from: https://github.com/sahana/eden/pull/1455

the template is built on initial understanding of the Sri Lanka UN-RCO shared requirements. This template serves to start the discussion on the developing the comprehensive [ShaRe blueprint](http://eden.sahanafoundation.org/wiki/BluePrint/ShaRe): 

Some minor inconsistencies persist and will be corrected in the next pull request: modules/templates/SHARE/Demo/tasks.cfg the "req" prepop is not working and the "doc" prepop is also throwing a warning; these data related bugs will be fixed

overall the template is working: http://www.sharehub.lk/ (you may use the default admin login for verifications)

### CODE REVIEW

**Comment by waidyanatha** [comment]
@nursix no rush but I plan to submit some enhancements; want to make sure this is acceptable first?

File: languages/ta.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# coding: utf8
```
Comment (nursix) [inline languages/ta.py]: Whilst this is not the standard format, it is still code to have a coding specification. The other language file has it:
https://github.com/sahana/eden/pull/1471/files#diff-62a3dd082e0cd5d87746918074a1103aR1

File: modules/templates/SHARE/Demo/sup_item_category.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Catalog,Category,Code,Asset,Telephone,Vehicle
+Equipment,Assets,Assets,True,,
```
Comment (nursix) [inline modules/templates/SHARE/Demo/sup_item_category.csv]: Number of columns doesn't match header row

File: modules/templates/SHARE/config.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        #    restricted = True,
+        #    module_type = 4
+        #)),
+        ("asset", Storage(
+            name_nice = "Assets",
+            #description = "Recording and Assigning Assets",
+            restricted = True,
+            module_type = 5,
+        )),
+        # Vehicle depends on Assets
```
Comment (nursix) [inline modules/templates/SHARE/config.py]: If you are not planning to activate these modules, then maybe you should remove them?

File: static/themes/SHARE/js/homepage.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+$(document).ready(function() {
```
Comment (nursix) [inline static/themes/SHARE/js/homepage.js]: This does nothing, and isn't included anywhere - remove this file?

Comment (nursix) [reply static/themes/SHARE/js/homepage.js]: ...and thereby probably the entire theme folder (no custom theme here?)

Comment (nursix) [reply static/themes/SHARE/js/homepage.js]: Meaning...this template does not (yet) have any custom theme, so it should not create a SHARE folder in static/themes either, especially not if the only JS file in it does nothing and is unused.

**Comment by waidyanatha** [comment]
What should be my strategy? Correct these first and do another PR? or can I submit these corrections along with the next PR? Pls advise.

**Comment by nursix** [comment]
You can fix these small issues with the next PR.

If you make more changes to this PR, though, please notify me for review.

**Comment by nursix** [comment]
Btw - what exactly is the need for this template to be in trunk at this point?

Since you can demo your draft from your own repo, it may not be necessary to merge it just yet - unless you're trying to get contributors from the community involved in its further development.

If you just want the code to be reviewed or are seeking technical advice, you can also post your requests to the mailing list together with a link to your repo. We can review and comment there as well.

**Comment by waidyanatha** [comment]
We (SSF) are working with the UN RCO in Sri Lanka. The contract is in 2 phases. In the first phase we are work with them on developing the [blueprint](http://eden.sahanafoundation.org/wiki/BluePrint/ShaRe) and test/demo data. In the second phase AidIQ will come in to develop the entire system. Since the amount for phase 1 is miniscule, I am refraining from bothering AidIQ and trying to build an off-the-shelf prototype to demo to them mainly to stimulate their thinking to capture the requirements. I thought if I did commit, then when it was time for AidIQ to step in, at least, the prepop, test/demo data, and a wireframe would be ready.

The issue is I have to switch between this SHARE project and two SAMBRO projects (Alerthub & PHIVOLCS). The SAMBRO projects definitely need to go into to trunk for users to pull. My github skills are not good enough to manage this three projects. Alternatively, I could start another eden fork to push the SHARE template work. But if you can bare with me for maybe, at most, 2 more PRs for the SHARE template, I should be done and AidIQ can take over?

**Comment by nursix** [comment]
I understand, and by no means intended to induce stress by forcing you to learn using story branches.

And yet - this doesn't seem to be the first time you're held up by pending PRs, and you wouldn't be if you were using story branches. You can even publish story branches if you want to share or collaborate on work that isn't meant for immediate submission as PR.

Keeping pending works separate until they really are trunk-ready, can make your life a little easier - so maybe you want to invest a little learning effort after all?

File: modules/templates/SHARE/config.py (hunk 3 above)
Comment (waidyanatha) [reply modules/templates/SHARE/config.py]: They are talking about logistics and I am not sure at this point whether we may or may not need these. I'm keeping these for now and if not required at the end will clean it up.

File: static/themes/SHARE/js/homepage.js (hunk 4 above)
Comment (waidyanatha) [reply static/themes/SHARE/js/homepage.js]: I have removed the JS folder. But I will still need the folder to hold the favicon and settings.ui.menu_logo image

**Pull request closed without merging**